tests/test_lib.py

```diff
@@ -896,3 +901,95 @@ def test_with_min_max_items_fails_validation(extra_tasks_data, input_data):
 
     with pytest.raises(vol.Invalid):
         validator(input_data)
+
+
+def test_required_any_of():
+    """Test schemas with Required(Any(...)) constraints."""
+    assert {
+        "properties": {
+            "color": {"type": "string"},
+            "temperature": {"type": "integer"},
+            "brightness": {"type": "integer"},
+        },
+        "required": [],
+        "type": "object",
+        "anyOf": [
+            {"required": ["color"]},
+            {"required": ["temperature"]},
+            {"required": ["brightness"]},
+        ],
+    } == convert(
+        {
+            vol.Required(vol.Any("color", "temperature", "brightness")): object,
+            vol.Optional("color"): str,
+            vol.Optional("temperature"): int,
+            vol.Optional("brightness"): int,
+        }
+    )
+
+    assert {
+        "properties": {
+            "color": {"type": "string"},
+            "temperature": {"type": "integer"},
+            "brightness": {"type": "integer"},
+            "mode": {"type": "string"},
+            "preset": {"type": "string"},
+        },
+        "required": [],
+        "type": "object",
+        "anyOf": [
+            {"required": ["color", "mode"]},
+            {"required": ["color", "preset"]},
+            {"required": ["temperature", "mode"]},
+            {"required": ["temperature", "preset"]},
+            {"required": ["brightness", "mode"]},
+            {"required": ["brightness", "preset"]},
+        ],
+    } == convert(
+        {
+            vol.Required(vol.Any("color", "temperature", "brightness")): object,
+            vol.Required(vol.Any("mode", "preset")): str,
+            vol.Optional("color"): str,
+            vol.Optional("temperature"): int,
+            vol.Optional("brightness"): int,
+            vol.Optional("mode"): str,
+            vol.Optional("preset"): str,
+        }
+    )
+
+    assert {
+        "properties": {
+            "entity_id": {"type": "string"},
+            "color": {"type": "string"},
+            "temperature": {"type": "integer"},
+        },
+        "required": ["entity_id"],
+        "type": "object",
+        "anyOf": [{"required": ["color"]}, {"required": ["temperature"]}],
+    } == convert(
+        {
+            vol.Required("entity_id"): str,
+            vol.Required(vol.Any("color", "temperature")): str,
+            vol.Optional("color"): str,
+            vol.Optional("temperature"): int,
+        }
+    )
+
+
+def test_required_any_of_description():
+    """Test that the description is preserved in a Required(Any(...)) constraint."""
+    assert {
+        "properties": {
+            "color": {"type": "string", "description": "Light appearance"},
+            "temperature": {"type": "string", "description": "Light appearance"},
+        },
+        "required": [],
+        "type": "object",
```

Comment:
can there be a description on the Required itself or on standalone `vol.Optional("color"): str,` field? I'm curious what those are supposed to do.